app/controllers/shared/api.php

```diff
@@ -114,6 +114,36 @@
 
     $deletes->setProject($project);
     $database->setProject($project);
+
+    $useCache = $route->getLabel('cache', false);
+    if ($useCache) {
+        $key = md5($request->getURI() . implode('*', $request->getParams()));
+        $cache = new Cache(new Filesystem(APP_STORAGE_CACHE . DIRECTORY_SEPARATOR . 'app-' . $project->getId()));
+        $timestamp = 60 * 60 * 24 * 30;
+        $data = $cache->load($key, $timestamp);
+        if (!empty($data)) {
+            $cacheLog = Authorization::skip(fn () => $dbForProject->getDocument('cache', $key));
+            if ($cacheLog->isEmpty()) {
+                Authorization::skip(fn () => $dbForProject->createDocument('cache', new Document([
+                '$id' => $key,
+                'accessedAt' => time(),
+                ])));
+            } else {
+                $cacheLog->setAttribute('accessedAt', time());
+                Authorization::skip(fn () => $dbForProject->updateDocument('cache', $cacheLog->getId(), $cacheLog));
+            }
+
+            $data = json_decode($data, true);
+            $response
+              ->addHeader('Expires', \date('D, d M Y H:i:s', \time() + $timestamp) . ' GMT')
+              ->addHeader('X-Appwrite-Cache', 'hit')
+              ->setContentType($data['content-type'])
+              ->file(base64_decode($data['payload']), $data['content-type'])
```

Comment:
Have we made sure that this causes the framework to skip the route execution? If not, the cache will not work.